src/auth_db/test.rs

```diff
@@ -26,25 +142,62 @@ fn get_bounces_invalid_address() {
 }
 
 #[test]
-fn deserialize_bounce_type() {
-    let bounce_type: BounceType = serde_json::from_value(json!(0)).expect("JSON error");
-    assert_eq!(bounce_type, BounceType::Soft);
-    let bounce_type: BounceType = serde_json::from_value(json!(1)).expect("JSON error");
-    assert_eq!(bounce_type, BounceType::Hard);
-    let bounce_type: BounceType = serde_json::from_value(json!(2)).expect("JSON error");
-    assert_eq!(bounce_type, BounceType::Soft);
-    let bounce_type: BounceType = serde_json::from_value(json!(3)).expect("JSON error");
-    assert_eq!(bounce_type, BounceType::Complaint);
-}
+fn create_bounce() {
+    let settings = Settings::new().expect("config error");
+    let db = DbClient::new(&settings);
 
-#[test]
-fn deserialize_invalid_bounce_type() {
-    match serde_json::from_value::<BounceType>(json!(4)) {
-        Ok(_) => assert!(false, "serde_json::from_value should have failed"),
-        Err(error) => assert_eq!(error.description(), "JSON error"),
-    }
-    match serde_json::from_value::<BounceType>(json!(-1)) {
-        Ok(_) => assert!(false, "serde_json::from_value should have failed"),
-        Err(error) => assert_eq!(error.description(), "JSON error"),
+    // The email addresses in this test are pseudo-namespaced so that we don't
+    // accidentally create bounce records that cause other tests to fail.
+    let bounces_before = db
+        .get_bounces("fxa-email-service.test.auth-db.foo@example.com")
+        .expect("db error");
+    let bounces_after =
+        db.create_bounce(
+            "fxa-email-service.test.auth-db.foo@example.com",
+            BounceType::Hard,
+            BounceSubtype::General,
+        ).and_then(|_| db.get_bounces("fxa-email-service.test.auth-db.foo@example.com"))
+            .expect("db error");
+    let now = now_as_milliseconds();
+
+    if bounces_before.len() == 20 {
```

Comment:
It's not at all clear where this magic number `20` comes from, or why.  I assume it's this:

  https://github.com/mozilla/fxa-auth-db-mysql/blob/dd249e5a2a75eca52f620fb8d2acc3e03e8393fd/lib/db/schema/patch-040-041.sql#L42

But that's along way removed for the reader to understand on sight.

Reply:
Also a good point, I'll add a comment.